sdcm/sct_config.py

```diff
@@ -1381,7 +1381,7 @@ class SCTConfiguration(dict):
     }
 
     multi_region_params = [
-        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'
+        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader', 'gce_datacenter'
```

Comment:
why are you adding this to this list ?, i.e. this shouldn't be here, 
those params are those who can have different value for different region, clearly `gce_datacenter` isn't that.

Reply:
seems like the unittest are failing cause of this change:
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-4978/3/execution/node/36/log/

Reply:
Am I misunderstanding the meaning of the list? In a multi dc scenario gce_datacenter will have "a different value" for each region? Same as region_name, which is in the list.
If gce_datacenter isn't in the multi_region_param list, the job fails nearly instantly:
```
Traceback (most recent call last):
  File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 1446, in <module>
    cli()
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 966, in run_test
    unittest.main(module=None, argv=['python -m unittest', argv],
  File "/usr/local/lib/python3.10/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/usr/local/lib/python3.10/unittest/main.py", line 147, in parseArgs
    self.createTests()
  File "/usr/local/lib/python3.10/unittest/main.py", line 158, in createTests
    self.test = self.testLoader.loadTestsFromNames(self.testNames,
  File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/local/lib/python3.10/unittest/loader.py", line 198, in loadTestsFromName
    inst = parent(name)
  File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 309, in __init__
    self.test_config.set_multi_region(len(self.params.get('gce_datacenter').split()) > 1)
AttributeError: 'list' object has no attribute 'split'
```

Reply:
`multi_region_params` was designed for AWS usage, and it used also for verification that all those field has the same length (a check which is done only for AWS, and adding this breaks it)

also a side effect of something being in this list, is that it turn back into a space separated string (to support all the code paths the assume those to be a string)

So:

1) there should be a separated list for each backend, for the verification sake
2) `gce_datacenter` should be turn back into a string, after validation.

Reply:
something like this:
```diff
diff --git a/sdcm/sct_config.py b/sdcm/sct_config.py
index f4550070..4e41ea47 100644
--- a/sdcm/sct_config.py
+++ b/sdcm/sct_config.py
@@ -621,7 +621,7 @@ class SCTConfiguration(dict):
 
         # GCE config options
 
-        dict(name="gce_datacenter", env="SCT_GCE_DATACENTER", type=str,
+        dict(name="gce_datacenter", env="SCT_GCE_DATACENTER", type=str_or_list,
              help="Supported: us-east1 - means that the zone will be selected automatically or "
                   "you can mention the zone explicitly, for example: us-east1-b"),
 
@@ -1389,10 +1389,10 @@ class SCTConfiguration(dict):
         "k8s-eks": [sct_abs_path('defaults/aws_config.yaml'), sct_abs_path('defaults/k8s_eks_config.yaml')],
     }
 
-    multi_region_params = [
-        'region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'
-    ]
-
+    _multi_region_params = {
+        "aws": ['region_name', 'n_db_nodes', 'ami_id_db_scylla', 'ami_id_loader'],
+        "gce": ['gce_datacenter', 'n_db_nodes'],
+    }
     stress_cmd_params = [
         # this list is used for variouse checks against stress commands, such as:
         # 1. Check if all c-s profile files existing that are referred in the commands
@@ -1423,6 +1423,7 @@ class SCTConfiguration(dict):
         if backend:
             backend_config_files += self.defaults_config_files[str(backend)]
 
+        self.multi_region_params = self._multi_region_params.get(str(backend), [])
         # 1) load the default backend config files
         files = anyconfig.load(list(backend_config_files))
         anyconfig.merge(self, files)
@@ -1755,8 +1756,8 @@ class SCTConfiguration(dict):
         db_type = self.get('db_type')
         self._check_version_supplied(backend)
         self._check_per_backend_required_values(backend)
-        if backend in ['aws'] and db_type != 'cloud_scylla':
-            self._check_aws_multi_region_params()
+        if backend in ['aws', 'gce'] and db_type != 'cloud_scylla':
+            self._check_multi_region_params()
 
         self._verify_data_volume_configuration(backend)
 
@@ -1798,8 +1799,9 @@ class SCTConfiguration(dict):
             if opt['name'] in self:
                 self._validate_value(opt)
 
-    def _check_aws_multi_region_params(self):
+    def _check_multi_region_params(self):
         region_count = {}
+        region_key = {"aws": 'region_name', 'gce': 'gce_datacenter'}
         for opt in self.multi_region_params:
             val = self.get(opt)
             if isinstance(val, str):
@@ -1808,7 +1810,7 @@ class SCTConfiguration(dict):
                 region_count[opt] = len(val)
             else:
                 region_count[opt] = 1
-        if not all(region_count['region_name'] == x for x in region_count.values()):
+        if not all(region_count[region_key.get(self.get('cluster_backend'))] == x for x in region_count.values()):
             raise ValueError("not all multi region values are equal: \n\t{}".format(region_count))
 
     def _validate_seeds_number(self):
diff --git a/unit_tests/test_config.py b/unit_tests/test_config.py
index 49c1ca8f..37a41165 100644
--- a/unit_tests/test_config.py
+++ b/unit_tests/test_config.py
@@ -576,6 +576,16 @@ class ConfigurationTests(unittest.TestCase):  # pylint: disable=too-many-public-
             conf = sct_config.SCTConfiguration()
             conf.verify_configuration()
 
+    def test_20_gce_mutli_dc(self):
+        os.environ['SCT_CLUSTER_BACKEND'] = 'gce'
+        os.environ['SCT_GCE_DATACENTER'] = 'a b c'
+        os.environ['SCT_N_DB_NODES'] = '2 2 2'
+        os.environ['SCT_GCE_IMAGE_DB'] = "image"
+        conf = sct_config.SCTConfiguration()
+        conf.verify_configuration()
+
+        self.assertEqual(conf.get('gce_datacenter'), 'a b c')
+
 
 if __name__ == "__main__":
     unittest.main()
```